File: practice/Mastermind/Mastermind.java
package practice.Mastermind;

import java.util.ArrayList;
import java.util.Scanner;

public class Mastermind {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // the number of clues
        int n = scanner.nextInt();
        scanner.nextLine(); // Consume the newline character

        ArrayList<Integer> remainingNumbers = new ArrayList<>();

        // Initialize the remaining numbers from 1 to 100
        for (int i = 1; i <= 100; i++) {
            remainingNumbers.add(i);
        }

        // Process the clues
        for (int i = 0; i < n; i++) {
            String guess_response = scanner.nextLine();
            String[] guess_response_split = guess_response.split(" ");
            int guess = Integer.parseInt(guess_response_split[0]);
            String response = guess_response_split[1];

            if (response.equalsIgnoreCase("Yes")) {
                // Keep only numbers divisible by the guess
                remainingNumbers.removeIf(num -> num % guess != 0);
            } else {
                // Remove numbers divisible by the guess
                remainingNumbers.removeIf(num -> num % guess == 0);
            }
        }

        // Output the first remaining number (the secret number)
        System.out.println(remainingNumbers.get(0));

        scanner.close();
    }
}

File: practice/Mastermind/Mastermind.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <algorithm>
using namespace std;

int main()
{
    int n;
    cin >> n;
    cin.ignore(); // Consume the newline character after the integer input

    vector<int> remainingNumbers;

    // Initialize the remaining numbers from 1 to 100
    for (int i = 1; i <= 100; i++)
    {
        remainingNumbers.push_back(i);
    }

    string guess_response;
    int guess;
    string response;

    // Process the clues
    for (int i = 0; i < n; i++)
    {
        getline(cin, guess_response);
        stringstream ss(guess_response);
        ss >> guess >> response;

        if (response == "Yes" || response == "yes")
        {
            remainingNumbers.erase(
                remove_if(remainingNumbers.begin(), remainingNumbers.end(),
                          [guess](int num)
                          { return num % guess != 0; }),
                remainingNumbers.end());
        }
        else
        {
            remainingNumbers.erase(
                remove_if(remainingNumbers.begin(), remainingNumbers.end(),
                          [guess](int num)
                          { return num % guess == 0; }),
                remainingNumbers.end());
        }
    }

    // Output the first remaining number (the secret number)
    cout << remainingNumbers[0] << endl;

    return 0;
}